### ISSUE TITLE
cannot set properties of a file with dash characters anymore

### ISSUE THREAD

**Issue by dudym** [devopshq/artifactory#250]
It seems like since https://github.com/devopshq/artifactory/commit/7e3180aaf8bf2f41d13c2a4e8cb33e13ecfaff9b (v0.7.605) we fail to archive files that have dash-characters in them:

```
	filename = 'myfile-with-dashes'
	username = 'user'
	password = 'pass'
	remote = "https://mydomainname/artifactory/my-location/"
	
	# Ref: 
	# https://pypi.python.org/pypi/artifactory
	# https://github.com/parallels/artifactory
	path = ArtifactoryPath(remote, auth=(username, password))

	# Upload
	path.deploy_file(filename)
	artifactoryAsset = ArtifactoryPath('/'.join([remote, filename]), auth=(username, password))
	artifactoryAsset.properties = properties
```
used to work.

Now we're getting
`
[Errno 2] No such file or directory: 'https://mydomainname/artifactory/my-location/api/storage/myfile-with-dashes'
`
@nacc - can you please have a look?

**Comment by nacc** [comment]
@dudym thank you for letting me know! I think I see the issue -- the underlying `requests.utils.quote` unconditionally quotes the '-' and does not provide an interface to override the underlying safe= parameter, to indicate the '-' in our case does not need to be escaped. I think this has to do with Artifactory's API actually being sort of non-standard. We my have to switch from using requests.quote to urllib.parse.quote directly.

That being said, we are regularly uploading and modifying properties of files (debs in our case) that have hyphens in them. I also confirmed our generated Docker image is using 0.7.605. So maybe there is something else at play.

I will try and reproduce this locally, can you confirm that undoing that change alone fixes it for you?

**Comment by nacc** [comment]
For what it's worth, I am getting a 404 even with 0.7.574, which I believe predates the referred to change. I'm debugging it still.

**Comment by nacc** [comment]
NM! It must have been some config option masking the underlying condition. Fix incoming shortly.

**Comment by nacc** [comment]
Sigh, now I'm thoroughly confused. I was able to (I think) reproduce your conditions with our Artifactory setup. But after trying to test my fix, when I go back and remove the files, have the admins empty the trash, etc. and re-try the original failing test case, it always works, with or without my fix. It feels like something must be being cached by the Artifactory backend, but I have no idea what. I'll keep digging, though. It is weird that my change would have altered any behavior, because the fix I see is that the REST API calls underlying deploy_file do not have any quoting currently (so I would expect them to be broken before and after my change). My change is at #252

**Comment by nacc** [comment]
@dudym would you be able to test with my proposed changes?

**Comment by zhan9san** [comment]
It may be fixed now. #254 
Could you try it again?

**Comment by beliaev-maksim** [comment]
@allburov please close as addressed by #254
